### ISSUE TITLE
[3.3.5] Succubus seduction and bladestorm

### ISSUE THREAD

**Issue by Raydor** [TrinityCore/TrinityCore#14273]
Core rev: https://github.com/TrinityCore/TrinityCore/commit/49ececf03830ca1d3df45bec1cd7f657d309450b

When Seduction is casted on a warrior that is casting Bladestorm, seduction gets resisted (immune) because warrior is immune to seduction while in bladestorm, the problem is that when immune pops up, the succubus keeps casting seduction even though the warrior was not affected by it. Seduction should be cancelled if the cast result is immune.

PS: It is only bugged with bladestorm, for example, with Divine Shield it doesn't get bugged.

Casting seduction on a warrior that is using bladestorm:
![](http://puu.sh/gib3B/3d753bd0c2.jpg)

It hits but the warrior is immune
![](http://puu.sh/gib3X/19aa647ef6.jpg)

The sucubus is still casting seduction and she doesn't stop until you give some order to her (like follow)
![](http://puu.sh/gib4N/2f281555ad.jpg)

The warrior is obviously not affected by seduction:
![](http://puu.sh/gib5F/c331b1c257.jpg)

**Comment by Runico** [comment]
I think this bug will happen on reflected too

**Comment by Raydor** [comment]
No, when warrior reflects Seduction, succubus gets seducted (I don't know if it should be like that because I think that if the Sucubbus is not channeling, the spell should not be affecting anyone and when Seduction gets reflected, she stops channeling).

**Comment by Rushor** [comment]
Do you have any source (blizzlike proof) for this?

**Comment by Rushor** [comment]
@Raydor: add a checkcast for the seduction spellscript and let it return false it the unit is immune to the spell

**Comment by Raydor** [comment]
Still bugged in https://github.com/TrinityCore/TrinityCore/commit/c210b935942f8f2202f1436f2cf73821614c041d, summoning @ariel-

**Comment by Keader** [comment]
@ariel- a little information...
When a spell get Immune on all effects, it should call hooks like: AfterHit, OnHit?
If yes, i think that it should be fixed by:

```diff
diff --git a/src/server/game/Spells/Spell.cpp b/src/server/game/Spells/Spell.cpp
index be01f88c2d..b7d19b7f67 100644
--- a/src/server/game/Spells/Spell.cpp
+++ b/src/server/game/Spells/Spell.cpp
@@ -2493,6 +2493,13 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
 
         CallScriptAfterHitHandlers();
     }
+
+    if (missInfo == SPELL_MISS_IMMUNE2)
+    {
+        m_caster->CombatStart(unit);
+        if (m_spellInfo->IsChanneled())
+            cancel();
+    }
 }
 
 SpellMissInfo Spell::DoSpellHitOnUnit(Unit* unit, uint32 effectMask, bool scaleAura)
```

**Comment by V3nomAnc3r** [comment]
Confirmed on rev https://github.com/TrinityCore/TrinityCore/commit/ebea1bf46ce9810580bc9c69bd1a6cbb6746dc26
